oteapi_dlite/strategies/filter.py

```diff
@@ -1,63 +1,156 @@
+    for removal.
 
-    - `("filterType", "dlite/create-collection")`
+    If `keep_referred` is true, any instance that is referred to by
+    an instance not marked for removeal is also unmarked for removal.
 
+    Finally, the instances that are still marked for removal are removed
+    from the collection.
     """
 
-    filter_config: FilterConfig
+    remove_label: str = Field(
+        None, description="Regular expression matching labels to remove."
+    )
+    remove_datamodel: str = Field(
+        None,
+        description="Regular expression matching datamodel URIs to remove.",
+    )
+    keep_label: str = Field(
+        None,
+        description="Regular expression matching labels to keep."
+        "This configuration overrides `regexp_remove` and `datamodel_remove`.  "
+        "Alias for the DLiteFilterStrategy `query` configuration.",
+    )
+    keep_datamodel: str = Field(
+        None,
+        description="Regular expression matching datamodel URIs to keep in "
+        "collection.  "
+        "This configuration overrides `regexp_remove` and `datamodel_remove`.",
+    )
+    keep_referred: bool = Field(
+        True,
+        description="Whether to keep all instances in the collection that are "
+        "directly or indirectly referred to (via ref-types or collections) "
+        "by kept instances.",
+    )
+
 
-    # Find a better way to keep collections alive!!!
-    # Need to be `Any`, because otherwise `pydantic` complains.
-    collection_refs: Dict[str, Any] = Field(
-        {},
-        description="A dictionary of DLite Collections.",
+class DLiteFilterConfig(FilterConfig):
+    """DLite generate strategy config."""
+
+    configuration: DLiteQueryConfig = Field(
+        ..., description="DLite filter strategy-specific configuration."
     )
 
-    def initialize(
-        self, session: "Optional[Dict[str, Any]]" = None
-    ) -> DLiteSessionUpdate:
-        """Initialize."""
-        if session is None:
-            raise ValueError("Missing session")
-        if "collection_id" in session:
-            raise KeyError("`collection_id` already exists in session.")
 
-        coll = dlite.Collection()
+@dataclass
+class DLiteFilterStrategy:
+    """Filter that removes all but specified instances in the collection.
+
+    The `query` configuration should be a regular expression matching labels
+    to keep in the collection.  All other labels will be removed.
 
-        # Make sure that collection stays alive
-        # It will never be deallocated...
-        coll._incref()  # pylint: disable=protected-access
+    **Registers strategies**:
 
-        # Store the collection in the data cache
-        cache = DataCache()
-        cache.add(value=coll.asjson(), key=coll.uuid)
+    - `("filterType", "dlite/filter")`
 
-        return DLiteSessionUpdate(collection_id=coll.uuid)
+    WARNING: This is a first simple implementation. The behaviour of this
+    strategy may change.
```

Comment:
Should this be implemented as a actual warning message maybe?

Reply:
Good question. The implementation has been updated since the warning was added, so maybe the warning is deprecated...

Reply:
Removed the warning